misc/dbt-materialize/dbt/adapters/materialize/connections.py

```diff
@@ -72,7 +72,17 @@ def connect(**kwargs):
 
 @dataclass
 class MaterializeCredentials(PostgresCredentials):
-    cluster: Optional[str] = "quickstart"
+    # NOTE(morsapaes) The cluster parameter defined in `profiles.yml` is not
+    # picked up in the connection string, but is picked up wherever we fall
+    # back to `target.cluster`. When no cluster is specified, either in
+    # `profiles.yml` or as a configuration, we should default to the default
+    # cluster configured for the connected dbt user (or, the active cluster for
+    # the connection). This is strictly better than hardcoding `quickstart` as
+    # the `target.cluster`, which might not exist and leads to all sorts of
+    # annoying errors. This will still fail if the defalt cluster for the
+    # connected user is invalid or set to `mz_catalog_server` (which cannot be
+    # modified).
+    cluster: Optional[str] = None
```

Comment:
Started looking into why the tests are failing, as far as I can tell so far the problem comes from here as the cluster name during the deploy tests is null for some reason:

```
unknown cluster '_dbt_deploy'
```

Still looking for a solution here.

Reply:
Yeah, that was exactly my conclusion. I _think_ we just need to jam the `deploy: True` variable somewhere else. Thanks for picking this up, @bobbyiliev!